### PR TITLE
Feature/ol 36 resource types

### PR DESCRIPTION
**Pull request by RajatMCAF**
[ADD] added resource type and new search criteria for resource availability

### CODE REVIEW

File: resource_reservation/models/resource_availability.py [hunk 1]
```diff
@@ -43,6 +45,84 @@ def search_availability(self):
...
+            'type': 'ir.actions.act_window',
+            'res_model': 'resource.reservation',
+            'view_mode': 'tree,form',
+            'domain': [('id', 'in', self.related_bookings.ids)],
+            'target': 'current',
+        }
+
+
+class ResourceAvailabilityByDate(models.Model):
+    _name = 'resource.availability.two'
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_availability.py:93]: change model name an related fields

File: resource_reservation/views/resource_availability.xml [hunk 2]
```diff
@@ -1,30 +1,84 @@
...
+                        <button name="show_related_bookings" string="Show Bookings" type="object" class="btn-primary"/>
+
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+
+            <record id="view_resource_availability_by_dates_form" model="ir.ui.view">
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_availability.xml:46]: align plz

File: resource_reservation/views/resource_availability.xml [hunk 3]
```diff
@@ -1,30 +1,84 @@
...
+
+                        <button name="search_availability" string="Check Availability" type="object" class="btn-primary"/>
+
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+        <record id="view_resource_availability_by_resource_form" model="ir.ui.view">
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_availability.xml:26]: align plz

File: resource_reservation/views/resource_details_views.xml [hunk 4]
```diff
@@ -19,7 +19,7 @@
     <field name="name">resource.type.tree</field>
     <field name="model">resource.type</field>
     <field name="arch" type="xml">
-        <tree name="Resource Types">
+        <tree name="Resource Types" editable="bottom">
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_details_views.xml:22]: why bottom if there are more fields

File: resource_reservation/views/resource_reservation_tag_views.xml [hunk 5]
```diff
@@ -22,7 +22,7 @@
         <field name="name">resource.reservation.tag.tree</field>
         <field name="model">resource.reservation.tag</field>
         <field name="arch" type="xml">
-            <tree>
+            <tree name="Reservation Tags" editable="bottom" >
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_tag_views.xml:25]: why bottom if there are more fields

File: resource_reservation/models/resource_availability.py [hunk 6]
```diff
@@ -43,6 +45,84 @@ def search_availability(self):
...
+        for reservation in self:
+            if reservation.start_datetime and reservation.end_datetime:
+                if reservation.end_datetime < reservation.start_datetime:
+                    raise exceptions.ValidationError(_("End date cannot"
+                                                       " be before the"
+                                                       " start date."))
+
+
+class ResourceAvailabilityByResource(models.Model):
+    _name = 'resource.availability.one'
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_availability.py:64]: why bottom if there are more fields

File: resource_reservation/views/resource_reservation_tag_views.xml (hunk 5 above)
Comment (RajatMCAF) [reply resource_reservation/views/resource_reservation_tag_views.xml:25]: I have changed it to "top"

File: resource_reservation/views/resource_details_views.xml (hunk 4 above)
Comment (RajatMCAF) [reply resource_reservation/views/resource_details_views.xml:22]: I have changed it to "top"

File: resource_reservation/views/resource_availability.xml (hunk 2 above)
Comment (RajatMCAF) [reply resource_reservation/views/resource_availability.xml:46]: I have aligned. Thank you

File: resource_reservation/views/resource_availability.xml (hunk 3 above)
Comment (RajatMCAF) [reply resource_reservation/views/resource_availability.xml:26]: I have aligned. Thank you

File: resource_reservation/models/resource_availability.py (hunk 6 above)
Comment (RajatMCAF) [reply resource_reservation/models/resource_availability.py:64]: I've changed it to "top"

File: resource_reservation/models/resource_availability.py (hunk 1 above)
Comment (RajatMCAF) [reply resource_reservation/models/resource_availability.py:93]: Done

**Pull request closed without merging**